pkg/clusterclientstore/store.go

```diff
@@ -303,13 +300,15 @@ func (s *Store) syncSecret(key string) error {
 	return s.create(clusterObj, secret)
 }
 
-func (s *Store) create(cluster *shipper.Cluster, secret *corev1.Secret) error {
-	checksum, ok := secret.GetAnnotations()[shipper.SecretChecksumAnnotation]
-	// Programmer error: this is filtered for at the informer level.
-	if !ok {
-		panic(fmt.Sprintf("Secret %q doesn't have a checksum annotation. this should be checked before calling 'create'", secret.Name))
-	}
+func computeSecretChecksum(secret *corev1.Secret) string {
+	hash := crc32.NewIEEE()
+	hash.Write(secret.Data["ca.crt"])
+	hash.Write(secret.Data["token"])
+	sum := hex.EncodeToString(hash.Sum(nil))
+	return sum
+}
```

Comment:
I wonder if we shouldn't compute the checksum of all of the secret's Data instead of just a few cherry-picked keys. And, if we do decide to just cherry-pick keys, maybe we should pick all of the possible ones Shipper uses to generate clients: https://github.com/bookingcom/shipper/blob/jgreff/app-targets/pkg/client/client.go#L40-L83

Reply:
@juliogreff good point, thanks. Spreading this logic of cherry-picking fragments of secrets seemed quite impractical to me, therefore it hashes the entire Data dataset now: https://github.com/bookingcom/shipper/pull/294/files#diff-dc16ca8ff134c0aed1343a896c6f9c53R453